Plugins/NativeCapture/Source/NativeCapture.cpp

```diff
@@ -4,15 +4,123 @@
+                frameProvider->ReadTextureAsync();
+                return gsl::finally<FrameProviderCleanup>([frameProvider{std::move(frameProvider)}]() mutable {
+                    frameProvider->m_cancellationToken.cancel();
+                });
+            }
+
+            ~OffScreenBufferFrameProvider()
+            {
+                bgfx::destroy(m_blitTextureHandle);
+            }
+
+        private:
+            OffScreenBufferFrameProvider(Babylon::Graphics::Impl& graphicsImpl, bgfx::FrameBufferHandle frameBufferHandle, FrameCallback callback)
+                : m_graphicsImpl{graphicsImpl}
+                , m_frameBufferTextureHandle{bgfx::getTexture(frameBufferHandle)}
+                , m_frameCallback{std::move(callback)}
+                , m_textureInfo{graphicsImpl.GetTextureInfo(m_frameBufferTextureHandle)}
+                , m_blitTextureHandle{bgfx::createTexture2D(m_textureInfo.Width, m_textureInfo.Height, m_textureInfo.HasMips, m_textureInfo.NumLayers, m_textureInfo.Format, BGFX_TEXTURE_BLIT_DST | BGFX_TEXTURE_READ_BACK)}
+            {
+                bgfx::TextureInfo textureInfo{};
+                bgfx::calcTextureSize(textureInfo, m_textureInfo.Width, m_textureInfo.Height, 1, false, m_textureInfo.HasMips, m_textureInfo.NumLayers, m_textureInfo.Format);
+                m_textureBuffer.resize(textureInfo.storageSize);
+            }
+
+            arcana::task<void, std::exception_ptr> ReadTextureAsync()
+            {
+                return arcana::make_task(m_graphicsImpl.AfterRenderScheduler(), m_cancellationToken, [thisRef{shared_from_this()}]{
+                    // bgfx does not allow readback of render textures, so the frame buffer render texture needs to be blitted to a texture with readback enabled.
+                    bgfx::blit(bgfx::getCaps()->limits.maxViews - 1, thisRef->m_blitTextureHandle, 0, 0, thisRef->m_frameBufferTextureHandle);
+
+                    // Reading the texture is an async operation, but everything that needs to be done prior to future write operations on that texture is completed synchronously,
+                    // so we kick off the read for the next frame prior to the read for the current frame completes.
+                    return arcana::when_all(thisRef->m_graphicsImpl.ReadTextureAsync(thisRef->m_blitTextureHandle, thisRef->m_textureBuffer).then(arcana::inline_scheduler, thisRef->m_cancellationToken, [thisRef]{
+                        thisRef->m_frameCallback(thisRef->m_textureInfo.Width, thisRef->m_textureInfo.Height, thisRef->m_textureInfo.Format, bgfx::getCaps()->originBottomLeft, thisRef->m_textureBuffer);
+                    }),
+                    thisRef->ReadTextureAsync()).then(arcana::inline_scheduler, arcana::cancellation::none(), [](const std::tuple<arcana::void_placeholder, arcana::void_placeholder>&){
```

Comment:
nit: since you don't care about the parameters, you could do this:
```suggestion
                    thisRef->ReadTextureAsync()).then(arcana::inline_scheduler, arcana::cancellation::none(), [](const auto&){
```